sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -144,6 +208,57 @@ public static IDisposable CreateHttpMessagePropertiesScope(IDictionary<string, o
             return CurrentHttpMessagePropertiesScope.Value;
         }
 
+        private ReadOnlyMemory<HttpPipelinePolicy> CreateRequestPipeline(HttpPipelinePolicy[] policies, List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)> customPolicies)
+        {
+            if (!_internallyConstructed)
+            {
+                throw new InvalidOperationException("Cannot send messages with per-request policies if the pipeline wasn't constructed with HttpPipelineBuilder.");
+            }
+
+            // Copy over client policies and splice in custom policies at designated indices
+            _pipeline.Slice(0, _perCallIndex).CopyTo(policies);
```

Comment:
It would be better to retrieve a span from _pipelines into a local at the beginning of this method and then use the span instead of using _pipelines. It's more efficient as span retrieval can be expensive.